src/core/eme/create_session.ts

```diff
@@ -0,0 +1,179 @@
+ * the given initializationData.
+ * @param {Uint8Array} initData
+ * @param {string} initDataType
+ * @param {Object} mediaKeysInfos
+ * @returns {Observable}
+ */
+export default function createSession(
+  initData: Uint8Array,
+  initDataType: string,
+  mediaKeysInfos: IMediaKeysInfos
+) : Observable<ICreateSessionEvent> {
+  return Observable.defer(() => {
+    const {
+      keySystemOptions,
+      mediaKeySystemAccess,
+      sessionsStore,
+      sessionStorage,
+    } = mediaKeysInfos;
+
+    const mksConfig = mediaKeySystemAccess.getConfiguration();
+    const sessionTypes = mksConfig.sessionTypes;
+    const hasPersistence = (
+      sessionTypes && arrayIncludes(sessionTypes, "persistent-license")
+    );
+
+    const sessionType : MediaKeySessionType =
+      hasPersistence &&
+      sessionStorage &&
+      keySystemOptions.persistentLicense ?
+      "persistent-license" : "temporary";
+
+    log.debug(`eme: create a new ${sessionType} session`);
+
+    const session = sessionsStore.createSession(initData, initDataType, sessionType);
+
+    // Re-check for Dumb typescript
+    if (!hasPersistence || !sessionStorage || !keySystemOptions.persistentLicense) {
+      if (__DEV__) {
+        assert(sessionType === "temporary");
+      }
+      return Observable.of({
+        type: "created-session" as "created-session",
+        value: { mediaKeySession: session, sessionType },
+      });
+    }
+    if (__DEV__) {
+      assert(sessionType === "persistent-license");
+    }
+
+    const storedEntry = sessionStorage.get(initData, initDataType);
+    if (!storedEntry) {
+      return Observable.of({
+        type: "created-session" as "created-session",
+        value: { mediaKeySession: session, sessionType },
+      });
+    }
+
+    /**
+     * Helper function to close and restart the current persistent session
+     * considered, and re-create it from scratch.
+     * @returns {Observable}
+     */
+    const recreatePersistentSession = () : Observable<INewSessionCreatedEvent> => {
+      log.info("eme: removing previous persistent session.");
+      if (sessionStorage.get(initData, initDataType) !== null) {
+        sessionStorage.delete(initData, initDataType);
+      }
+      return sessionsStore.closeSession(session)
+        .map(() => {
+          const newSession =
+            sessionsStore.createSession(initData, initDataType, sessionType);
+          return {
+            type: "created-session" as "created-session",
+            value: { mediaKeySession: newSession, sessionType },
+          };
+        });
+    };
+
+    return loadPersistentSession(storedEntry.sessionId, session)
+      .mergeMap((hasLoadedSession) : Observable<ICreateSessionEvent> => {
+        if (!hasLoadedSession) {
+          log.warn("eme: no data stored for the loaded session");
+          sessionStorage.delete(initData, initDataType);
+          return Observable.of({
+            type: "created-session" as "created-session",
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4325))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=212&issues=AWNFqf_5XOBV_YrhBEpQ&open=AWNFqf_5XOBV_YrhBEpQ)

[ISSUE_KEY]: AWNFqf_5XOBV_YrhBEpQ